core/ir/x86/decode_table.c

```diff
@@ -7450,632 +7439,1029 @@ const instr_info_t vex_W_extensions[][2] = {
     {OP_vmovd, 0x660f6e10, "vmovd", Vdq, xx, Ed, xx, xx, mrm|vex, x, tvexw[109][0]},
     {OP_vmovq, 0x660f6e50, "vmovq", Vdq, xx, Ey, xx, xx, mrm|vex, x, tvexw[109][1]},
   }, { /* vex_W_ext 109 */
-    {OP_vmovd, 0x660f7e10, "vmovd", Ed, xx, Vd_dq, xx, xx, mrm|vex, x, tevexw[135][0]},
-    {OP_vmovq, 0x660f7e50, "vmovq", Ey, xx, Vq_dq, xx, xx, mrm|vex, x, tevexw[135][1]},
+    {OP_vmovd, 0x660f7e10, "vmovd", Ed, xx, Vd_dq, xx, xx, mrm|vex, x, tevexw[136][0]},
+    {OP_vmovq, 0x660f7e50, "vmovq", Ey, xx, Vq_dq, xx, xx, mrm|vex, x, tevexw[136][2]},
   },
 };
 
 /****************************************************************************
  * Instructions that differ depending on evex.W.
  * Index is evex.W value
  */
-const instr_info_t evex_W_extensions[][2] = {
+const instr_info_t evex_W_extensions[][4] = {
   {    /* evex_W_ext 0 */
-    {OP_vmovups, 0x0f1010,"vmovups", Ves,xx,KEd,Wes,xx,mrm|evex|ttfvm,x,tevexw[1][0]},
+    {OP_vmovups, 0x0f1000,"vmovups", Ves,xx,KEd,Wes,xx,mrm|evex|ttfvm,x,tevexw[1][0]},
+    {INVALID, 0x0f1010,"(bad)", xx,xx,xx,xx,xx,no,x,NA},
```

Comment:
So this opcode really becomes invalid if evex.b is set?

Reply:
Yes. The EVEX prefixed vmovups belongs to exception class E4.nb which #UD if EVEX.b=1.

Reply:
OK.